assembly-ide/src/assembly/bin/che.sh

```diff
@@ -10,20 +10,436 @@
+1. Open browser to http://${host}:${USE_PORT}, or:
+2. Open native chromium app.
+#############################################################\n\n"
+
+  fi
+}
+
+function call_catalina {
+
+  # Test to see that Che application server is where we expect it to be
+  if [ ! -d "${ASSEMBLY_BIN_DIR}" ]; then
+    error_exit
+    return
+  fi
+
+  ### Cannot add this in setenv.sh.
+  ### We do the port mapping here, and this gets inserted into server.xml when tomcat boots
+  [ -z "${JAVA_OPTS}" ]  && export JAVA_OPTS="-Dport.http=${USE_PORT}"
+  #Class path
+  [ -z "${SERVER_PORT}" ]  && export SERVER_PORT=${USE_PORT}
+
+  # Launch the Che application server, passing in command line parameters
+  ${ASSEMBLY_BIN_DIR}/catalina.sh ${USE_SERVER_ACTION}
+
+}
+
+function stop_che_server {
+
+  if ! $USE_DOCKER; then
+    echo -e "Stopping Che server running on localhost:${USE_PORT}"
+    call_catalina >/dev/null 2>&1
+  else
+    echo -e "Stopping Che server running in docker container."
+
+    DOCKER_EXEC="sudo service docker stop && /home/user/che/bin/che.sh stop"
+    "${DOCKER}" exec che $DOCKER_EXEC >/dev/null 2>&1
+    DOCKER_EXIT=$?
+
+    echo -e "Stopping docker container named che."
+    "${DOCKER}" stop che >/dev/null 2>&1
+    DOCKER_EXIT=$?
+  fi
+}
+
+function launch_che_server {
+
+ strip_url $DOCKER_HOST
+ print_client_connect
+
+  # Launch Che natively as a tomcat server
+  if ! $USE_DOCKER; then
+    call_catalina
+
+  # Launch Che as a docker image
+  else
+    
+    # Check to see if the Che docker was not properly shut down
+    "${DOCKER}" inspect che &> /dev/null
+    DOCKER_INSPECT_EXIT=$?
+
+    echo -e "Starting Che server in existing docker container named che."
+
+    # Attempt restart of existing container named "che"
+    "${DOCKER}" start che >/dev/null 2>&1
+    DOCKER_EXIT=$?
+    
+    DOCKER_EXEC="sudo service docker start && /home/user/che/bin/che.sh start" >/dev/null 2>&1
+    "${DOCKER}" exec che $DOCKER_EXEC 
+    #sudo service docker start && cd /home/user/che/bin && ./che.sh start
+
+    # If either command fails, then wipe any existing image and restarting
+    if [ ${DOCKER_INSPECT_EXIT} -eq 1 ] || [ ${DOCKER_EXIT} -eq 1 ]; then
+      echo "Either che container does not exist, or duplicate conflict was discovered."
+      echo -e "Removing any old containers and launching a new one using image: codenvy/che:${USE_DOCKER_TAG}..."
+      "${DOCKER}" kill che &> /dev/null
+      "${DOCKER}" rm che &> /dev/null
+
+      if $win || $mac ; then
+        "${DOCKER}" run --privileged -e '"'DOCKER_MACHINE_HOST=${host}'"' --name che -it -p ${USE_PORT}:${USE_PORT} -p 32768-32788:32768-32788 codenvy/che:${USE_DOCKER_TAG} #&> /dev/null
+      else
+        "${DOCKER}" run --privileged --name che -it -p ${USE_PORT}:${USE_PORT} -p 32768-32788:32768-32788 codenvy/che:${USE_DOCKER_TAG} &> /dev/null
+      fi    
+    fi
+  fi
+}
+
+
+# Call function
+parse_command_line $@
```

Comment:
Added them in quotes.

If user does not pass any command line arguments, then defaults will be used: port:8080, "run" action, and it will launch Che natively, not in a docker image.